### PR TITLE
Bug 1833419: Detect WebhookDescription changes in CSVs

### PR DESCRIPTION
**Pull request by awgreene**
Problem:
Once a CSV has reached the Succeeded phase, updating
WebhookDescriptions in the CSV on cluster does not cause OLM to update
existing ValidatingWebhookConfigurations or
MutatingWebhookConfigurations on cluster.

Solution:
Update OLM to calculate if a change to the WebhookDescription has
occurred when a CSV is updated and if so update the on cluster
resources.

### CODE REVIEW

File: pkg/controller/install/webhook.go [hunk 1]
```diff
@@ -163,6 +168,7 @@ func (i *StrategyDeploymentInstaller) createOrUpdateValidatingWebhook(ogNamespac
 }
 
 const WebhookDescKey = "webhookDescriptionGenerateName"
+const WebhookHashKey = "webhookDescriptionSHA"
```
Comment (ecordell) [inline pkg/controller/install/webhook.go:171]: We should probably align with the naming convention for `pod-template-hash` (see the dep hash label too)

File: pkg/controller/operators/olm/apiservices.go [hunk 2]
```diff
@@ -486,3 +486,34 @@ func (a *Operator) updateDeploymentSpecsWithApiServiceData(csv *v1alpha1.Cluster
 	}
 	return strategyDetailsDeployment, nil
 }
+
+func (a *Operator) areWebhooksAvailable(csv *v1alpha1.ClusterServiceVersion) (bool, error) {
```
Comment (ecordell) [inline pkg/controller/operators/olm/apiservices.go:490]: Won't this be an issue if there's an update to a webhook that changes the name of the webhook? I think we'll be left with an "old" webhook, right?

File: pkg/controller/install/webhook.go (hunk 1 above)
Comment (awgreene) [reply pkg/controller/install/webhook.go:171]: Understood, I'll make the following update:
```
const WebhookHashKey = "olm.webhook-description-hash"
```

File: pkg/controller/operators/olm/apiservices.go (hunk 2 above)
Comment (awgreene) [reply pkg/controller/operators/olm/apiservices.go:490]: Yep - I'll update the PR to handle this case and make sure that the added e2e test checks for this.

File: pkg/controller/install/webhook.go (hunk 1 above)
Comment (awgreene) [reply pkg/controller/install/webhook.go]: Updated in the latest version of the PR.

File: pkg/controller/operators/olm/apiservices.go (hunk 2 above)
Comment (awgreene) [reply pkg/controller/operators/olm/apiservices.go:539]: Updated in the latest version of the PR.

File: pkg/controller/operators/olm/operator.go [hunk 3]
```diff
@@ -1721,13 +1721,14 @@ func (a *Operator) checkReplacementsAndUpdateStatus(csv *v1alpha1.ClusterService
 func (a *Operator) updateInstallStatus(csv *v1alpha1.ClusterServiceVersion, installer install.StrategyInstaller, strategy install.Strategy, requeuePhase v1alpha1.ClusterServiceVersionPhase, requeueConditionReason v1alpha1.ConditionReason) error {
 	apiServicesInstalled, apiServiceErr := a.areAPIServicesAvailable(csv)
 	strategyInstalled, strategyErr := installer.CheckInstalled(strategy)
+	webhooksInstalled, webhookErr := a.areWebhooksAvailable(csv)
```
Comment (ecordell) [inline pkg/controller/operators/olm/operator.go:1724]: should this happen after checking `strategyErr`?

File: test/e2e/webhook_e2e_test.go [hunk 4]
```diff
@@ -26,7 +26,7 @@ const (
 	webhookName = "webhook.test.com"
 )
 
-var _ = Describe("CSVs with a Webhook", func() {
+var _ = FDescribe("CSVs with a Webhook", func() {
```
Comment (ecordell) [inline test/e2e/webhook_e2e_test.go:29]: need to remove focus to merge

**Pull request merged**